evaporate.js

```diff
@@ -885,7 +886,7 @@
                 }
                 var completedAt = new Date(u.completedAt || FAR_FUTURE);
 
-                return con.partSize === u.partSize && completedAt > HOURS_AGO;
+                return con.partSize === u.partSize && completedAt > HOURS_AGO && me.name === u.awsKey && con.bucket === u.bucket;
```

Comment:
I think we need to make sure we're describing the same thing. We're using what you describe to be a bug -- at the rate of 100K + uploads per month. EvaporateJS does not upload to a different location -- and by that I mean, bucket. Given that the "right" way to be performant with S3 is to create randomized file names, the implemented feature uses heuristics (from the uploader's perspective) to save uploading costs for large files. The frequency with which uploads fail for batch file uploading of big files is surprisingly large, thus the optimization.

You only get the optimization if you enable it explicitly. It's disabled by default.

If you still want to use the feature with your restrictions, then I'm happy to consider an enhancement that enriches or restricts the existing functionality without breaking usage. For example, in your case, you want to say, "hey, only optimize if the uploaded object name === S3 object name". That's definitely something that would be an enhancement, disabled by default, and enabled by you in your use cases.